lib/eex/lib/eex/compiler.ex

```diff
@@ -34,67 +42,88 @@ defmodule EEx.Compiler do
   end
 
   defp generate_buffer([{:expr, line, mark, chars} | rest], buffer, scope, state) do
-    expr = Code.string_to_quoted!(chars, [line: line, file: state.file])
+    expr = Code.string_to_quoted!(chars, line: line, file: state.file)
     buffer = state.engine.handle_expr(buffer, IO.chardata_to_string(mark), expr)
     generate_buffer(rest, buffer, scope, state)
   end
 
   defp generate_buffer([{:start_expr, start_line, mark, chars} | rest], buffer, scope, state) do
     {contents, line, rest} = look_ahead_text(rest, start_line, chars)
+
     {contents, rest} =
-      generate_buffer(rest, state.engine.handle_begin(buffer), [contents | scope],
-                      %{state | quoted: [], line: line, start_line: start_line})
+      generate_buffer(rest, state.engine.handle_begin(buffer), [contents | scope], %{
+        state | quoted: [], line: line, start_line: start_line
```

Comment:
Please do this update on its own, like this

```elixir
new_state = %{state | quoted: [], ...}
generate_buffer(rest, state.engine.handle_begin(buffer), [contents | scope], new_state)
```

There's another case where we should do this below :)